include/libtorrent/extensions.hpp

```diff
@@ -214,6 +219,10 @@ namespace libtorrent
 		// called when plugin is added to a session
 		virtual void added(session_handle) {}
 
+		// called after a plugin is added
+		// allows the plugin to register DHT requests it would like to handle
+		virtual void register_dht_extensions(dht_extensions_t&) {}
```

Comment:
is dht_extensions_t documented anywhere? (it's just a vector of strings -> handler mappings, right?). It might make sense to document how to use this here. Also, dht_extensions_t does not need to be exported, right?